content/docs/typechecking-with-proptypes.md

```diff
@@ -30,105 +30,107 @@ Greeting.propTypes = {
 };
 ```
 
-`PropTypes` exports a range of validators that can be used to make sure the data you receive is valid. In this example, we're using `PropTypes.string`. When an invalid value is provided for a prop, a warning will be shown in the JavaScript console. For performance reasons, `propTypes` is only checked in development mode.
+`PropTypes` експортує ряд валідаторів, які можуть бути використані щоб впевнетись, що ви отримали вірні дані. В наведеному вище прикладі ми викостивуємо `PropTypes.string`. Якщо якийсь проп отримає невірне значення, в консолі  JavaScript буде показано попередження. З міркувань продуктивності `propTypes` перевіряються лише в режимі розробки.
 
 ### PropTypes {#proptypes}
 
-Here is an example documenting the different validators provided:
+Приклад використання наявних валідаторів:
 
 ```javascript
 import PropTypes from 'prop-types';
 
 MyComponent.propTypes = {
-  // You can declare that a prop is a specific JS type. By default, these
-  // are all optional.
+  // Ви можете декларувати, що проп має вказаний JS тип.
+  // Ці типи за замовченням дозволяють відсутність значення.
   optionalArray: PropTypes.array,
   optionalBool: PropTypes.bool,
-  optionalFunc: PropTypes.func,
+  optionalFunc: PropTypes.func, // наприклад, приймає функцію або відсутнє значення
   optionalNumber: PropTypes.number,
   optionalObject: PropTypes.object,
   optionalString: PropTypes.string,
   optionalSymbol: PropTypes.symbol,
 
-  // Anything that can be rendered: numbers, strings, elements or an array
```

Comment:
Please keep the translation within 2 lines so we won't have any merge conflicts in future.